src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -448,20 +475,24 @@
         this)
       (a/stop [this]
         (shutdown! app-context)
-        this))))
+        this)
+      (a/restart [this]
+        (run-lifecycle-fns app-context s/stop "stop" (reverse ordered-services))
```

Comment:
I've just pushed out a new commit that wraps the restart in a try-catch to achieve this effect.